crypto/armcap.c

```diff
@@ -112,20 +112,23 @@ static unsigned long getauxval(unsigned long key)
  * ARM puts the feature bits for Crypto Extensions in AT_HWCAP2, whereas
  * AArch64 used AT_HWCAP.
  */
+# ifndef AT_HWCAP
```

Comment:
Should these not be under `#ifdef __linux__`? Otherwise you still risk the same thing happening on FreeBSD if something gets screwed up and sys/auxv.h isn't included.

Reply:
I don't think __linux__ would be correct. The use is later guarded by OSSL_IMPLEMENT_GETAUXVAL which is only set on __FreeBSD_version >= 1200000, but, it does raise the question about android etc.

Reply:
I think the current state of the PR is commensurate with what OpenSSL does already in analogous situations.  Yes, there may be some edge cases where it breaks, but we are going to have to rely on user reports to find those cases.

Reply:
I do think `#ifdef __linux__` would be appropriate.  The AT_* values are OSABI-specific and could vary on other operating systems.  Note that on FreeBSD it was merely luck that 16 happened to be AT_CANARY and that AT_CANARY happened to always fail to be read due to its special nature vs returning random bits.  A mismatch on another OS might instead result in parsing a user space address or some other field like a user ID as the HWCAP bit mask.